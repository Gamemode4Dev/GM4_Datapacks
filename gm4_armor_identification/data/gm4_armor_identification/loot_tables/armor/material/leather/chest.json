{
  "type": "minecraft:generic",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:leather_chestplate",
          "functions": [
            {
              "function": "minecraft:set_attributes",
              "modifiers": [
                {
                  "attribute": "generic.armor",
                  "name": "generic.armor",
                  "amount": 3,
                  "operation": "addition",
                  "slot": "chest"
                }
              ]
            },
            {
              "function": "minecraft:set_name",
              "entity": "this",
              "name": [
                {
                  "translate": "%1$s%3427655$s",
                  "with": [
                    {
                      "text": "Unidentified ",
                      "italic": false
                    },
                    {
                      "translate": "item.gm4.mob_buff.armor.name.unidentified"
                    }
                  ]
                },
                {
                  "translate": "item.minecraft.leather_chestplate",
                  "italic": false
                }
              ]
            },
            {
              "function": "minecraft:set_nbt",
              "tag": "{gm4_armor_identification:{unidentified:0,slot:1,name:'{\"italic\":false,\"translate\":\"item.minecraft.leather_chestplate\"}'}}"
            },
            {
              "function": "minecraft:enchant_with_levels",
              "levels": {
                "min": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ai_data",
                  "scale": 1.5
                },
                "max": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ai_data",
                  "scale": 2.5
                }
              },
              "treasure": false,
              "conditions": [
                {
                  "condition": "minecraft:value_check",
                  "value": {
                    "type": "minecraft:score",
                    "target": {
                      "type": "minecraft:fixed",
                      "name": "$difficulty"
                    },
                    "score": "gm4_ai_data"
                  },
                  "range": {
                    "min": 2
                  }
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.1
                }
              ]
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:depth_strider"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_armor_identification:mobs/modifiers/reef"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.5
                }
              ]
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:projectile_protection"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_armor_identification:mobs/modifiers/growth"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.8
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
