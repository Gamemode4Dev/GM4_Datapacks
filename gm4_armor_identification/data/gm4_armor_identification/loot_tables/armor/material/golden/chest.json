{
  "type": "minecraft:generic",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:golden_chestplate",
          "functions": [
            {
              "function": "minecraft:set_attributes",
              "modifiers": [
                {
                  "attribute": "generic.armor",
                  "name": "generic.armor",
                  "amount": 5,
                  "operation": "addition",
                  "slot": "chest"
                }
              ]
            },
            {
              "function": "minecraft:set_name",
              "entity": "this",
              "name": [
                {
                  "translate": "%1$s%3427655$s",
                  "with": [
                    {
                      "text": "Unidentified ",
                      "italic": false
                    },
                    {
                      "translate": "item.gm4.armor_identification.armor.name.unidentified"
                    }
                  ]
                },
                {
                  "translate": "item.minecraft.golden_chestplate",
                  "italic": false
                }
              ]
            },
            {
              "function": "minecraft:set_nbt",
              "tag": "{gm4_armor_identification:{unidentified:0,slot:1,name:'{\"italic\":false,\"translate\":\"item.minecraft.golden_chestplate\"}'}}"
            },
            {
              "function": "minecraft:enchant_with_levels",
              "levels": {
                "min": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ai_data",
                  "scale": 2
                },
                "max": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ai_data",
                  "scale": 3
                }
              },
              "treasure": false,
              "conditions": [
                {
                  "condition": "minecraft:value_check",
                  "value": {
                    "type": "minecraft:score",
                    "target": {
                      "type": "minecraft:fixed",
                      "name": "$difficulty"
                    },
                    "score": "gm4_ai_data"
                  },
                  "range": {
                    "min": 2
                  }
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.33
                }
              ]
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:depth_strider"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_armor_identification:mobs/modifiers/reef"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.5
                }
              ]
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:projectile_protection"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_armor_identification:mobs/modifiers/growth"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.8
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
